Feature/462 help installation (#467)
memeLab/Jandig: docs/installation-guide.md

## Step 1 - Cloning the Repository

The first step after downloading the prerequisites is clone the repository, so we are going to need the following code:
```
git clone https://github.com/memeLab/Jandig
cd Jandig
```
If you are having any problems with the git commands consider checking the Github documentation for problems with the [cloning](https://docs.github.com/pt/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors) or your user [authentication](https://docs.github.com/pt/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)

## Step 2 - Running the Docker

Jandig uses docker and docker-compose to ensure a consistent development environment and to make the deploy process as painless as possible, so we are going to initialize the docker and then run the docker-compose.yml
```
docker-compose -f docker/docker-compose.yml up
```

### ERROR - Env file not found

> If you receive an ERROR message saying it couldn't find env file, you probably didn't create the .env file yet. So navigate to **Jandig\src\\.envs\\** and create a file named **.env** and copy the content of **.example** and paste it in this file. Then run the above code again.

Reviewer: ```suggestion
If you receive an error message saying it couldn't find env file, you probably didn't create the .env file yet. So navigate to **Jandig\src\\.envs\\** and create a file named **.env** and copy the content of **.example** and paste it in this file. Then run the above code again.
```

### ERROR - System could't find CreateFile

> If you are receiving the following ERROR, that means your Docker has problems to initiate. Try to initialize the Docker Desktop manually with admin privileges, you can check the Docker Desktop running in your toolbar. If your docker isn't open yet consult the [DockerDesktop](https://docs.microsoft.com/pt-br/windows/wsl/tutorials/wsl-containers) documentation.

Reviewer: ```suggestion
If you are receiving the following error using Windows and Docker, that means your Docker has problems to initiate. Try to initialize the Docker Desktop manually with admin privileges, you can check the Docker Desktop running in your toolbar. If your docker isn't open yet consult the [Docker Desktop](https://docs.microsoft.com/pt-br/windows/wsl/tutorials/wsl-containers) documentation.
```

![env-file](./images/installation-guide-createFile.png)

> ### ERROR - Duplicated dependency
> If you are receiving the following ERROR or a similar one, that means that there are duplicated depencies. Sometimes when new Pull Requests are merged in the development branch, some dependencies can be updated or removed, and that results in duplicate dependencies or missing ones. This may not appear to other Jandig developers, since they already set their environments. 

Reviewer: This error happened only once and will be avoided after we add Poetry to Jandig. I'm not sure this one makes sense to be added, it isn't a 'common' error for new contributors.
```suggestion

```

### ERROR - Minio
If you encounter a screen as the image below, or some error message related to Minio, it's very likely that the `USE_MINIO` in **.env** file is set to True. To correct this, set it to False (`USE_MINIO=False`).

![env-file](./images/installation-guide-minio.png)

### ERROR - Migration

To solve the backend issue with migrations (image below)

![env-file](./images/installation-guide-migration.png)

do the following steps, as @GustavoAPS commented [here](https://github.com/memeLab/Jandig/issues/436#issuecomment-1034420328):
1. Run the container.
1. Open a new terminal and list the containers with "docker ps"
1. Enter the project container with "docker exec -it docker_django_1 bash"
1. Inside the container make the migrations with: `python3 src/ARte/manage.py migrate`
1. After the migrations, the problem should be solved.

> 

Reviewer: Jandig implemented Invoke with some "recipes" to make our lives as developers easier with a simple `inv command`. To run migrations on our containers you could use:
```bash
cd docker/
docker-compose run django inv db
# or if you already have a django container running
docker-compose exec django inv db
```
